WooCommerce/Classes/Authentication/AuthenticationManager.swift

```diff
@@ -263,36 +263,35 @@ extension AuthenticationManager: WordPressAuthenticatorDelegate {
     /// Presents the Login Epilogue, in the specified NavigationController.
     ///
     func presentLoginEpilogue(in navigationController: UINavigationController, for credentials: AuthenticatorCredentials, onDismiss: @escaping () -> Void) {
-        let matcher = ULAccountMatcher()
-        matcher.refreshStoredSites()
 
-        /// Jetpack is required. Present an error if we don't detect a valid installation for a self-hosted site.
-        if let urlFromCredentials = credentials.wpcom?.siteURL ?? credentials.wporg?.siteURL,
-           isJetpackValidForSelfHostedSite(url: urlFromCredentials) {
-            return presentJetpackError(for: urlFromCredentials, with: credentials, in: navigationController, onDismiss: onDismiss)
+        guard let siteURL = credentials.wpcom?.siteURL ?? credentials.wporg?.siteURL else {
+            // This should not happen since the resulting credentials should be either `wpcom` or `wporg`
+            return DDLogError("⛔️ No site URL found to present Login Epilogue.")
         }
 
-        // We are currently supporting WPCom credentials only
-        // Update this when handling store credentials authentication.
-        guard let wpcomLogin = credentials.wpcom else {
-            return
+        /// Jetpack is required. Present an error if we don't detect a valid installation for a self-hosted site.
+        if isJetpackValidForSelfHostedSite(url: siteURL) {
+            return presentJetpackError(for: siteURL, with: credentials, in: navigationController, onDismiss: onDismiss)
```

Comment:
any reasons it's returning an "error" state while the condition indicates that it's a "valid" state?

Reply:
it looks like `isJetpackValidForSelfHostedSite` function name might be misleading, since the code seems to return if it's "invalid." Maybe we can update the function name?

Reply:
Great catch! I didn't notice this in the last PR 😅 Updated in bebc20b8d31edaabaeead7f7a52f5d6da80fe8f2.